app/services/PayeAtsService.scala

```diff
@@ -46,19 +48,46 @@ class PayeAtsService @Inject()(middleConnector: MiddleConnector, auditService: A
         Left(HttpResponse(INTERNAL_SERVER_ERROR, e.getMessage))
     }
 
-  def getPayeATSMultipleYearData(nino: Nino, yearFrom: Int, yearTo: Int)(
+  def getPayeTaxYearData(nino: Nino, yearFrom: Int, yearTo: Int)(
     implicit hc: HeaderCarrier,
-    request: PayeAuthenticatedRequest[_]): Future[Either[HttpResponse, List[PayeAtsData]]] =
+    request: AuthenticatedRequest[_]): Future[Either[HttpResponse, List[Int]]] =
     middleConnector.connectToPayeATSMultipleYears(nino, yearFrom, yearTo) map { response =>
-      handleConnectorResponse[List[PayeAtsData]](response, nino, yearFrom)
+      handlePayeTaxYearDataResponse[List[PayeAtsData]](response, nino, yearFrom)
     } recover {
       case e: BadRequestException => Left(HttpResponse(BAD_REQUEST, e.getMessage))
-      case e: NotFoundException   => Left(HttpResponse(NOT_FOUND, e.getMessage))
+      case e: NotFoundException   => Right(List.empty)
       case e: Exception =>
         Logger.error(s"Exception in PayeAtsService: $e", e)
         Left(HttpResponse(INTERNAL_SERVER_ERROR, e.getMessage))
     }
 
+  private def handlePayeTaxYearDataResponse[A](response: HttpResponse, nino: Nino, taxYear: Int)(
+    implicit reads: Reads[List[PayeAtsData]],
+    hc: HeaderCarrier,
+    request: AuthenticatedRequest[_]): Either[HttpResponse, List[Int]] =
+    response.status match {
+      case OK =>
+        sendAuditEventForPayeTaxYearData(nino, taxYear)
```

Comment:
flatmap this future or you will hit a race condtion when testing it

Reply:
removed that line as I removed the audit method